### ISSUE TITLE
How to pass new pre-trained embeddings while sharing the same vocabulary across torchtext.Field?

### ISSUE THREAD

**Issue by jacopo-repossi** [pytorch/text#1013]
## ❓ Questions and Help

**Description**
I was trying to concatenate two embedding layers in my CNN from two different pre-trained embeddings, before applying my convolutions.

Here's the basic workflow:

    # Load pre-trained embeddings (dim=100)
    from torchtext.vocab import Vectors
    vectors_amazon = Vectors(name='gensim_embeddings_amazon.txt', cache='{}/{}'.format(PROJECT_FOLDER, OUTPUT_FOLDER))
    vectors_imdb = Vectors(name='gensim_embeddings_imdb.txt', cache='{}/{}'.format(PROJECT_FOLDER, OUTPUT_FOLDER))

Then I create my `text_field` and `label_field` as follow:

    # Custom_tokenizer is my tokenizer, MAX_SIZE=20000
    text_field = Field(sequential=True, use_vocab=True, tokenize=custom_tokenizer)
    label_field = LabelField()

    text_field.build_vocab(train_data,
                           max_size=MAX_SIZE,
                           vectors=vectors_amazon)
    label_field.build_vocab(train_data)

and after creating my train/valid/test iterator with `BucketIterator`, I create my CNN `model` for sentiment analysis.

So far so good, the problem is that I'd like to create another embedding considering also the `vectors_imdb` and I'm stuck here, since for the embedding layer I'd do the following:

    pretrained_embeddings = text_field.vocab.vectors
    model.embedding.weight.data.copy_(pretrained_embeddings)

but I have no idea how I can pass to a `model.second_embedding.weight.data` the values in `vectors_imdb` while keeping the correct alignment between embeddings and sharing the same vocab (coming from my training data)

I tried something like 

    second_text_field = text_field
    second_text_field.vocab.set_vectors(text_field.vocab.stoi, vectors_imdb, 100)

but of course it doesn't work since changing vectors in `second_text_field` also modify `text_field` ones.

How can I modify vectors while keeping the correct mapping word:vector representation?
I'm afraid the only way is to loop through `vectors_imdb` keeping only words that are in my vocab, sorting them so that the two embeddings match and pass the result to the second_embedding layer, right?

**Comment by zhangguanheng66** [comment]
First of all, we are going to retire `Field` class so we recommend to switch away from it. Here is an [example](https://github.com/pytorch/text/blob/master/examples/data_pipeline/pipelines.py#L133-L138) to write a vector-based data pipeline. We are also working on an end-to-end tutorial to show how to use `torchtext.experimental.transforms`.

**Comment by jacopo-repossi** [comment]
Thank you for the reply and your time.
However it's not quite clear to me yet how I can solve my problem: how can I load another pre-trained embedding given that one has already been created with `build_vocab` (so I have to align each word:vector representation)?

**Comment by jacopo-repossi** [comment]
Any suggestion?

Thanks for your time

**Comment by zhangguanheng66** [comment]
> Thank you for the reply and your time.
> However it's not quite clear to me yet how I can solve my problem: how can I load another pre-trained embedding given that one has already been created with `build_vocab` (so I have to align each word:vector representation)?

You are right that the vocab is associated with the embedding. Can you load your vocab and assign tokens to the corresponding ids?
